plan/newplanbuilder.go

```diff
@@ -694,14 +737,16 @@ func (b *planBuilder) buildNewSelect(sel *ast.SelectStmt) LogicalPlan {
 		b.err = errors.New("SelectLocal have not been supported!")
 		return nil
 	}
-	if hasAgg {
-		if sel.GroupBy != nil {
-			p, correlated, gbyCols = b.resolveGbyExprs(p, sel.GroupBy, sel.Fields.Fields)
-			if b.err != nil {
-				return nil
-			}
+	if sel.GroupBy != nil {
+		p, correlated, gbyCols = b.resolveGbyExprs(p, sel.GroupBy, sel.Fields.Fields)
+		if b.err != nil {
+			return nil
 		}
-		aggFuncs, havingMap, orderMap, totalMap = b.extractAggFunc(sel)
+	}
+
+	havingMap, orderMap = b.resolveHavingAndOrderBy(sel, p)
```

Comment:
Add comments here to describe why resolve having/orderby clause before extracting aggFuncs.